src/main/java/net/masterthought/cucumber/Trends.java

```diff
@@ -30,10 +31,69 @@
 
     private long[] durations = new long[0];
 
+    private  FeatureScenario[][] featuresDetail = new FeatureScenario[0][0];
+
     public String[] getBuildNumbers() {
         return buildNumbers;
     }
 
+    public FeatureScenario[][] getFeaturesDetail() {
+    return featuresDetail;
+    }
+
+    public ArrayList<TrendTableRow> collectTrendFeatureScenario() {
+
+        ArrayList<TrendTableRow> trendTableRows = new ArrayList<>();
+
+        for (int i = 0 ; i < buildNumbers.length;i++){
+
+            for (int j=0;j<featuresDetail[i].length;j++){
+                FeatureScenario featureScenario = featuresDetail[i][j];
+                TrendTableRow trendTableRow = new TrendTableRow(featureScenario.getDeviceName(), featureScenario.getFeatureName(), featureScenario.getScenarioName(), featureScenario.getId());
+                if(!isAlreadyInTrenTableRows(trendTableRow,trendTableRows)){
+                    trendTableRows.add(trendTableRow);
+                }
+            }
+        }
+
+        for (int i = 0 ; i < buildNumbers.length;i++){
+            FeatureScenario[] featureScenarioPerBuild = featuresDetail[i];
+            for (TrendTableRow trendTableRow: trendTableRows) {
+                if (getStatus(trendTableRow, featureScenarioPerBuild)!=null){
+                    trendTableRow.setStatus(getStatus(trendTableRow, featureScenarioPerBuild));
+                }else {
+                    trendTableRow.setStatus("-");
+                }
+            }
+        }
+
+        return trendTableRows;
+    }
+
+    private boolean isAlreadyInTrenTableRows(TrendTableRow trendTableRow, ArrayList<TrendTableRow> TrendTableRows){
+        boolean found = false;
+        for (TrendTableRow ttr: TrendTableRows) {
+            if(ttr.getId().equals(trendTableRow.getId())){
+                found = true;
+                break;
+            }
+        }
+
+        return found;
+    }
+
+    private String getStatus(TrendTableRow trendTableRow, FeatureScenario[] featureScenarios){
+        String status = null;
+        for (int i=0; i < featureScenarios.length ; i++){
+            String status_tmp = featureScenarios[i].getStatus();
+            if (trendTableRow.getId().equals(featureScenarios[i].getId())){
+                status = status_tmp;
```

Comment:
if I'm correct result of this method depends of `featureScenarios` order - was it intended?

Reply:
in Trend table we will have only one report (feature, scenario, array status per build) even there are more than one report due to retry failed scenario. we will take status from last retry. 

for example, we have
Feature: Login
Scenario: Login with correct credentials
The first run, test failed. Then we retry it and get status passed. In this case, in trend table we only have 1 record with last status
Feature: Login
Scenario: Login with correct credentials
Status: Passed

To check that feature and scenario is same, we use id that contain from deviceId, featureName, scenarioName, and line